File: LeetCode/121_Best_Time_To_buy_And_Sell_Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 1) return 0;

        int minPrice = numeric_limits<int>::max();
        int maxProfit = 0;

        for (int i = 0; i < prices.size(); ++i) {
            minPrice = min(minPrice, prices[i]);
            maxProfit = max(maxProfit, prices[i] - minPrice);
        }
        return maxProfit;
    }
};

File: LeetCode/121_Best_Time_To_Buy_And_Sell_Stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut min_price = 10000;
        let mut max_profit = 0;

        for price in prices {
            max_profit = std::cmp::max(max_profit, price - min_price);
            min_price = std::cmp::min(min_price, price);
        }
        max_profit
    }
}